numpy/lib/_iotools.py

```diff
@@ -506,13 +506,15 @@ class StringConverter:
     _mapper.extend([(nx.float64, float, nx.nan),
                     (nx.complex128, complex, nx.nan + 0j),
                     (nx.longdouble, nx.longdouble, nx.nan),
-                    (nx.unicode_, asunicode, '???'),
-                    (nx.string_, asbytes, '???'),
                     # If a non-default dtype is passed, fall back to generic
                     # ones (should only be used for the converter)
                     (nx.integer, int, -1),
                     (nx.floating, float, nx.nan),
-                    (nx.complexfloating, complex, nx.nan + 0j),])
+                    (nx.complexfloating, complex, nx.nan + 0j),
```

Comment:
In what circumstance are these actually used? What fails if you just remove these three rather than moving two to the bottom.

(out of scope for this PR, but curious

Reply:
They should b be used for other dtypes such as `i4` or `float16` which are not specifically listed here. I guess the real reason why not to list all of them and rather have these is that it was the smaller change. (there are test failures if you delete this), but only for the `integer` case.